### PR TITLE
Improve MM2 Grafana dashboard

### PR DESCRIPTION
**Pull request by fvaleri**
This commit adds new panels which are useful to see if the mirroring pipeline is working fine or needs tuning.

- Fix "Number of Tasks" with decimal point by setting decimal to zero in general options
- Add "Source Connector - Connect Outstanding Messages Queue"
- Add "Source Connector - Producer Available Buffer"
- Add "Source Connector - Average Offset Commit Time"

When dealing with high volumes of messages, a high producer buffer or low offset commit timeout may cause MM2 to fail with "Failed to flush, timed out while waiting for producer to flush outstanding X messages". For that reason, we are interested in the following metrics: 

- `kafka_connect_source_task_source_record_active_count` (Connect's outstanding messages queue)
- `kafka_producer_buffer_available_bytes` (producer's available buffer, default: 32MiB) 
- `kafka_connect_connector_task_offset_commit_avg_time_ms` (offset commit timeout, default: 5s)

### CODE REVIEW

**Comment by scholzj** [comment]
Can you share screenshots with what changed in the dashabord?

File: packaging/examples/metrics/grafana-dashboards/strimzi-kafka-mirror-maker-2.json [hunk 1]
```diff
@@ -20,19 +20,31 @@
       "type": "panel",
       "id": "graph",
       "name": "Graph",
-      "version": "5.0.0"
+      "version": ""
```
Comment (scholzj) [inline packaging/examples/metrics/grafana-dashboards/strimzi-kafka-mirror-maker-2.json:23]: Can you explain the different changes outside of the new panels? Or try to minimize them if they are not intentional?

Comment (fvaleri) [reply packaging/examples/metrics/grafana-dashboards/strimzi-kafka-mirror-maker-2.json:23]: So I just imported the dashboard from main branch, added the 3 new panels and exported from the sharing menu (external export). I also used the same Grafana version that is on main (7.3.7). I also tried to import it back successfully. Is this procedure correct?

Comment (scholzj) [reply packaging/examples/metrics/grafana-dashboards/strimzi-kafka-mirror-maker-2.json:23]: Well, TBH when I do some changes, given I have no clue abotu the meanings of the feilds which it decides to change automatically I usually manually revert them back. That gives it a good changce that I do not break anything with my PR.

**Comment by fvaleri** [comment]
Attaching screenshots with the proposed metrics:

### MM2 idle
idle

### MM2 working
working

**Comment by scholzj** [comment]
So what exactly do these charts mean and how are they expected to be used?

**Comment by fvaleri** [comment]
> So what exactly do these charts mean and how are they expected to be used?

These charts monitor Connect's outstanding messages queue size, total producer's available buffer and the average offset commit time related of MirrorSourceConnector, which are important for MM2 tuning as explained below.

Let's say a user with high volumes or large backlog of messages starts a single MM2 instance with default settings. At this point they will likely hit:

```
2022-02-07 13:10:34,717 ERROR WorkerSourceTask{id=source->target.MirrorSourceConnector-0} Failed to flush, timed out while waiting for producer to flush outstanding 898458 messages (org.apache.kafka.connect.runtime.WorkerSourceTask) [SourceTaskOffsetCommitter-1]
2022-02-07 13:10:34,765 ERROR WorkerSourceTask{id=source->target.MirrorSourceConnector-0} Failed to commit offsets (org.apache.kafka.connect.runtime.SourceTaskOffsetCommitter) [SourceTaskOffsetCommitter-1]
```

Connect's outstanding messages queue can grow indefinitely (there is no hard limit). This causes the producer's buffer to fill up and the (blocking) flush operation to last longer and longer, until it eventually times out.

Often, increasing the `offset.flush.timeout.ms` is not sufficient, and you also need to reduce `producer.buffer.memory` in order to keep Connect's outstanding messages queue under reasonable size. All of this assuming we do not want to raise MM2 resources, and we are fine with the resulting throughput.

**Review by scholzj — APPROVED** [review]
Ok, I did the cleanup of the changes for you. That should make it easier to track the changes.

**Review by ppatierno — APPROVED** [review]
Very useful addition, thanks!

**Comment by scholzj** [comment]
Thanks for the PR @fvaleri

**Pull request merged**